### DISCUSSION TITLE
how exp claim works?

### DISCUSSION THREAD

**Discussion by bokunodev** [panva/paseto#46] [Q&A]
question regarding `exp` and `nbf` claim on paseto v2
[v2-paseto-protocol-version-v2](https://github.com/panva/paseto/blob/main/docs/README.md#v2-paseto-protocol-version-v2)

`exp` and `nbf` claim is a duration is string format. how does that work? does that mean, paseto token never expired if never used?

**Comment by Jacobinoo** [comment]
`exp` is when token expires, whereas `nbf` identifies the time before which the token must NOT be accepted. This is useful when you want to prepare a token that’s not yet valid until the specified time/day etc.
